deepsampler-persistence/src/main/java/de/ppi/deepsampler/persistence/bean/ext/MapPrimitiveKeyExtension.java

```diff
@@ -0,0 +1,131 @@
+        try {
+            return instantiateMap(originalBeanClass);
+        } catch (PersistenceException e) {
+            // isProcessable() should limit this extension to objects that implement the Map interface.
+            // In some special cases the concrete Map cannot be instantiated without complex problems.
+            // E.G. this is the case if originalBean has been created by Collections.unmodifiableMap().
+            // Such methods return objects of inner classes that are invisible (private class) or don't have a default constructor,
+            // so there is no simple way to instantiate them here. In cases like this, we fall back to a normal HashMap, even though
+            // this changes the persisted Bean.
+
+           return new HashMap<>();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<Object, Object> instantiateMap(Class<?> persistentBeanClass) {
+        try {
+            Constructor<Map<Object, Object>> defaultConstructor = (Constructor<Map<Object, Object>>) persistentBeanClass.getConstructor();
+            return defaultConstructor.newInstance();
+        } catch (NoSuchMethodException | InstantiationException | IllegalAccessException | InvocationTargetException e) {
+            throw new PersistenceException("The targetType %s cannot be instantiated.", e, persistentBeanClass.getTypeName());
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> T revert(Object persistentBean, Class<T> targetClass, ParameterizedType type, PersistentBeanConverter persistentBeanConverter) {
+        Type[] genericParameterTypes = type.getActualTypeArguments();
+
+        if (genericParameterTypes.length != 2) {
+            throw new PersistenceException("%s is only able to deserialize to Map<String, V>. But we try to deserialize %s",
+                    getClass().getTypeName(), targetClass.getTypeName());
+        }
+
+        Class<T> valueClass;
+        ParameterizedType valueType;
+
+        if (genericParameterTypes[1] instanceof ParameterizedType) {
+            valueType = (ParameterizedType) genericParameterTypes[1];
+            valueClass = (Class<T>) valueType.getRawType();
+        } else {
+            valueType = null;
+            valueClass = (Class<T>) genericParameterTypes[1];
+        }
+
+        Class<?> keyClass = (Class<T>) genericParameterTypes[0];
+
+        Map<Object, Object> valueMap = instantiateMap(persistentBean.getClass());
+
+        ((Map<String, Object>) persistentBean).entrySet().stream()
+                .map(entry -> {
+                    Object key = ReflectionTools.parseString(entry.getKey(), keyClass);
+                    Object value = persistentBeanConverter.revert(entry.getValue(), valueClass, valueType);
+                    return new Object[]{key, value};
+                })
+                .forEach(pair -> valueMap.put(pair[0], pair[1]));
+
+        return (T) valueMap;
```

Comment:
I think a conventional for each would be more readable, multi-line lambda + entryMap stream is too much in my opinion.